### PR TITLE
Add secured etcd support

### LINKED ISSUE #21: Etcd DB Conf and Certs
Hi @kvaps,

I'm looking to add ability to connect to a etcd cluster.
At the moment, we can do :
```yaml
controller:
  db:
    connectionUrl: "etcd://node1:2379,node2:2379"
```
But etcd is secured with TLS certs so, currently, we can't make a secure connection with your helm chart.
Before making a PR, I want to know if you will be OK with my solution.

Add 4 config values, and by the way we can set etcd prefix :
```yaml
controller:
  db:
    tls: true
    cert: [A base64 encoded PEM format certificate]
    key: [A base64 encoded PEM format private key]
    ca: [A base64 encoded PEM format certificate authority]
    # optional
    etcdPrefix: "/LINSTOR/"
```

Create `kubernetes.io/tls` secret `db-tls.yaml` :
```yaml
{{- $fullName := include "linstor.fullname" . -}}
{{- if .Values.controller.enabled }}
{{- if .Values.controller.db.tls }}
---
apiVersion: v1
kind: Secret
metadata:
  name: {{ $fullName }}-db-tls
  annotations:
    "helm.sh/resource-policy": "keep"
    "helm.sh/hook": "pre-install"
    "helm.sh/hook-delete-policy": "before-hook-creation"
    "directives.qbec.io/update-policy": "never"
type: kubernetes.io/tls
data:
  tls.crt: {{ .Values.controller.db.cert }}
  tls.key: {{ .Values.controller.db.key }}
  ca.crt: {{ .Values.controller.db.ca }}
{{- end }}
{{- end }}
```

Modify `_helpers.tpl` to add reference to certs :
```
[...]
  connection_url = "{{ .Values.controller.db.connectionUrl }}"
{{- if .Values.controller.db.tls }}
  ca_certificate = "/config/ssl/db/ca.crt"
  client_certificate = "/config/ssl/db/tls.crt"
  client_key_pkcs8_pem = "/config/ssl/db/tls.key"
{{- end }}
{{- if .Values.controller.db.etcdPrefix }}
  [db.etcd]
  prefix = "{{ .Values.controller.db.etcdPrefix }}"
{{- end }}
[...]
```

Modify `controller-deployment.yaml` :
```yaml
{{- $fullName := include "linstor.fullname" . -}}
{{- if .Values.controller.enabled }}
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: {{ $fullName }}-controller
  name: {{ $fullName }}-controller
  namespace: {{ .Release.Namespace }}
[...]
      initContainers:
      - name: load-certs
[...]
        command:
[...]
          {{- end }}
          {{- if .Values.controller.db.tls }}
          cp -rf /tls/db /config/ssl/db
          {{- end }}
          rm -f "$tmp"
        volumeMounts:
        {{- if .Values.controller.db.tls }}
        - name: db-tls
          mountPath: /tls/db
        {{- end }}
[...]
      volumes:
      {{- if .Values.controller.db.tls }}
      - name: db-tls
        secret:
          secretName: {{ $fullName }}-db-tls
      {{- end }}
[...]

### PR DESCRIPTION
**Pull request by jbanety**
ref https://github.com/kvaps/kube-linstor/pull/22
fixes #21

### CODE REVIEW

File: helm/kube-linstor/templates/_helpers.tpl [hunk 1]
```diff
@@ -31,6 +31,15 @@ Generates linstor.toml config file
...
   connection_url = "{{ .Values.controller.db.connectionUrl }}"
+{{- if .Values.controller.db.tls }}
+  ca_certificate = "/tls/db/ca.crt"
+  client_certificate = "/tls/db/tls.crt"
+  client_key_pkcs8_pem = "/tls/db/tls.key"
+{{- end }}
+{{- if .Values.controller.db.etcdPrefix }}
+  [db.etcd]
+  prefix = "{{ .Values.controller.db.etcdPrefix }}"
+{{- end }}
```
Comment (kvaps) [inline helm/kube-linstor/templates/_helpers.tpl]:
```suggestion
{{- with .Values.controller.db.etcdPrefix }}
  [db.etcd]
  prefix = "{{ . }}"
{{- end }}
```

**Review by kvaps — COMMENTED** [review]
Thanks!
Just little correction, I think using `with` syntax is more correct in this case

**Comment by kvaps** [comment]
values.yaml changes are missing 😕

**Comment by jbanety** [comment]
> values.yaml changes are missing 😕

Fixed

**Comment by kvaps** [comment]
merged, thanks a lot!🎉

**Pull request merged**